### DISCUSSION TITLE
Dynamic server-side page not re-rendering

### DISCUSSION THREAD

**Discussion by jonathatargino** [vercel/next.js#53233] [Help]
### Summary

Im developing a simple project to manage contacts. In a server-side page, i fetch user's contacts and send it to a component via prop.
![image](https://github.com/vercel/next.js/assets/102263444/2bf63de8-d3b5-42de-adf1-c26115cd618d)

![image](https://github.com/vercel/next.js/assets/102263444/4c5074a1-55b9-4564-87c6-3592d7ef37c6)
When user clicks in add contacts (next Link component) and goes to contacts form to add new contact:
![image](https://github.com/vercel/next.js/assets/102263444/25c3ef6b-7882-4fb0-b8f2-63a10142bbba)
![image](https://github.com/vercel/next.js/assets/102263444/dc0eafa8-567d-4fe4-acb0-b582090635a3)
Going back contacts page, the contacts list fetched from backend isn't updated.
![image](https://github.com/vercel/next.js/assets/102263444/374dee89-97f6-4fe5-bd2e-60ee76dcd87b)
![image](https://github.com/vercel/next.js/assets/102263444/eacf19f3-1a02-403e-a544-e41ada43f68a)

Also, these fetch have cache value to "no-cache". I've tryied "no-store" too but didn't worked for me.

How do i fix it? I want to re-fetch the data again when users goes to /contacts

According the next docs "In a dynamic route, components are rendered on the server at request time.". Knowing this, my /contacts page have to be rendered on the server at request time, no ? the page have dynamic function like cookies() from next/headers

  

### Example

https://my-contacts-frontend.vercel.app/

**Comment by icyJoseph** [comment]
https://github.com/vercel/next.js/discussions/51488#discussioncomment-6216934

**Reply by jonathatargino** [reply]
So, to force the data from contacts to update, I have to make a function using revalidatePath, like this:
![image](https://github.com/vercel/next.js/assets/102263444/8cc9d911-9c99-4eec-96f4-5db15494f5d0)
To use after every function that will mutate my contats, alright?
update function usage: (onSuccess of my ContactForm component react query's mutation)
![image](https://github.com/vercel/next.js/assets/102263444/fa2c20ae-dbc2-49f6-80db-b01b694db56a)

I have setted the experimental serverActions to true in next config too
![image](https://github.com/vercel/next.js/assets/102263444/36168f67-9884-413f-9164-8c01cc7cd716)

but the function updateContacts seems to have no effect
